src/effects/Echo.cpp

```diff
@@ -196,16 +196,25 @@ struct EffectEcho::Validator
 
    void PopulateOrExchange(ShuttleGui& S, const EffectSettings& settings);
 
-   EffectEchoSettings& mSettings;
+   EffectEchoSettings mSettings;
```

Comment:
The steps aren't exactly as I described before, but that is all right.

Did you verify that each step in the transformation preserves the persistency of settings?

Maybe the second and fourth steps could be combined as the third step of three.

Because you never construct the Validator in the second commit -- it isn't necessary to make mSettings a reference and then change it to a value in the last step.